apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx

```diff
@@ -104,23 +87,26 @@ export const TokensTableRow = ({
 
     // TODO: make a separate hook out of this and add GA
     try {
-      setApproving(true)
-      const summary = `Approve ${tokenData?.symbol || 'token'}`
-      await approve({ modalMessage: summary, transactionSummary: summary })
+      openApproveModal(tokenData?.symbol)
+      await approveCallback(`Approve ${tokenData?.symbol || 'token'}`)
     } catch (error: any) {
       console.error(`[TokensTableRow]: Issue approving.`, error)
       handleSetError(error?.message)
     } finally {
-      setApproving(false)
+      closeApproveModal()
     }
-  }, [account, approve, handleCloseError, handleSetError, toggleWalletModal, tokenData?.symbol])
-
-  const isApproved = approvalState === ApprovalState.APPROVED
-  const isPendingOnchainApprove = approvalState === ApprovalState.PENDING
-  const isPendingApprove = !isApproved && (approving || isPendingOnchainApprove)
+  }, [
+    account,
+    approveCallback,
+    handleCloseError,
+    handleSetError,
+    toggleWalletModal,
+    tokenData?.symbol,
+    openApproveModal,
+    closeApproveModal,
+  ])
 
   const hasZeroBalance = !balance || balance?.equalTo(0)
-  const hasNoAllowance = !currentAllowance || currentAllowance.equalTo(0)
```

Comment:
I assume this behaviour is covered but the approval state?

Reply:
Yep, you are right